src/Modules/AuthModule/components/Register/Register.tsx

```diff
@@ -1,5 +1,403 @@
-import React from "react";
+import React, { useContext, useState } from "react";
+import { ErrorMessage } from "@hookform/error-message";
+import axios from "axios";
+import { useForm } from "react-hook-form";
+import { Link, useNavigate } from "react-router-dom";
+import { toast } from "react-toastify";
+
+import Images from "../../../ImageModule/components/Images/Images";
+
+import BaceUrlContext from "../../../../Context/BaceUrlContext";
 
 export default function Register() {
-  return <>Register</>;
+  // instance from use navigate
+  const navigate = useNavigate();
+  // image input state
+  const [image, setImage] = useState(null);
+  // hide pass or show state
+  let [hidePassInInpt, setHidePassInInpt] = useState(true);
+  // check loading state
+  const [isLoading, setIsLoading] = useState(false);
+  // function for change password input type
+  const changePassInputType = () => {
+    setHidePassInInpt((hidePassInInpt = !hidePassInInpt ? true : false));
+  };
+  const btnloading = () => {
+    return (
+      <div className="loader">
+        <i>&lt;</i>
+        <span>LOADING</span>
+        <i>/&gt;</i>
+      </div>
+    );
+  };
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    watch,
+  } = useForm({
+    criteriaMode: "all",
+  });
+  const onSubmit = async (data: any) => {
+    setIsLoading(true);
+    const registerFormData = appendToFormData(data);
+    try {
+      const response = await axios.post(
+        `https://upskilling-egypt.com:3003/api/v1/Users/Register`,
+        registerFormData
+      );
+      toast.success(response.data.message);
+      // navigate("/verifyaccount");
+      console.log(response.data.message);
+    } catch (err) {
+      console.log(err);
+      toast.error(err.response.data.message);
+    }
+    setIsLoading(false);
+  };
+  const appendToFormData = (data: any) => {
```

Comment:
Instead of having this function in every form handling, we can make a general utility function that will handle  any formData given and return it.
something like that:
```js
const appendToFormData = (data: any) => {
    const formData = new FormData();
    
    Object.keys(data).forEach(key => {
        if (data[key] !== null && data[key] !== undefined) {
            formData.append(key, data[key]);
        }
    });

    return formData;
};

```